extensions/ql-vscode/src/model-editor/yaml.ts

```diff
@@ -34,38 +41,92 @@ function createDataProperty(
     .join("\n")}`;
 }
 
+function createExtensions<T>(
+  language: QueryLanguage,
+  methods: readonly T[],
+  definition: ModelsAsDataLanguagePredicate<T> | undefined,
+) {
+  if (!definition) {
+    return "";
+  }
+
+  return `  - addsTo:
+      pack: codeql/${language}-all
+      extensible: ${definition.extensiblePredicate}
+    data:${createDataProperty(methods, definition)}
+`;
+}
+
 export function createDataExtensionYaml(
   language: QueryLanguage,
   modeledMethods: readonly ModeledMethod[],
 ) {
   const modelsAsDataLanguage = getModelsAsDataLanguage(language);
 
-  const methodsByType: Record<
-    Exclude<ModeledMethodType, "none">,
-    ModeledMethod[]
-  > = {
-    source: [],
-    sink: [],
-    summary: [],
-    neutral: [],
-  };
+  const methodsByType = {
+    source: [] as SourceModeledMethod[],
+    sink: [] as SinkModeledMethod[],
+    summary: [] as SummaryModeledMethod[],
+    neutral: [] as NeutralModeledMethod[],
+  } satisfies Record<keyof ModelsAsDataLanguagePredicates, ModeledMethod[]>;
```

Comment:
I've never seen the [`satisfies` operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html) before but it looks useful.

Reply:
It's quite new, but it's perfect here to ensure that all keys are included, and all values are assignable to `ModeledMethod[]`.